src/gcl/MeshVelocityAlg.C

```diff
@@ -79,8 +79,7 @@ MeshVelocityAlg<AlgTraits>::MeshVelocityAlg(Realm& realm, stk::mesh::Part* part)
     NUM_IP, isoParCoords_, isoCoordsShapeFcnHostView_.data());
   Kokkos::deep_copy(isoCoordsShapeFcnDeviceView_, isoCoordsShapeFcnHostView_);
 
-  auto scsFaceNodeMapHostView =
-    Kokkos::create_mirror(scsFaceNodeMapDeviceView_);
+  Kokkos::View<int[12][4], Kokkos::HostSpace> scsFaceNodeMapHostView;
```

Comment:
Doesn't the change from create_mirror to a direct definition reverse a previous commit?  I wonder if int[12][4] is unstable across platforms and should be int** instead. But if this gets Frontier up and running, great catch.

Reply:
Yes, I agree with these questions. It would seem like `[12][4]` is specific to a single element/face topology. Is this algorithm only going to run on a single kind of elements/faces? Generally `create_mirror` is considered safer because it automatically creates the same type/shape as the device view.
But like James says, if this makes it work then it's hard to argue against.

Reply:
Looking at the CI failure indicates something is amiss here.

Reply:
Do we have any unit tests for this?

Reply:
If this is causing a problem in running the Challenge Problem, how about moving the const int scsFaceNodeMap_[12][4] array from the header and into the one place it is used inside the execute() function, replacing Kokkos::View scsFaceNodeMap with this const int array.  Just avoid copying it over to device at all since it is const data anyway.

Reply:
What's really odd is that this only seems to be a problem when using FSI.

Reply:
There should be a way to query the size and then loop over the dims correctly. This is host cost after all.

Reply:
This algorithm gets used only when FSI is activated.

Reply:
Yes the test that is failing CI.

Reply:
If James' solution of making it into a const int array is feasible, that sounds good.
But I agree with Paul, it should be easy to query the sizes and then that should guarantee correct looping. You can get those sizes with `.extent(0)` and `.extent(1)`, I think. Would need to verify exact syntax...

Reply:
@psakievich Might be worthwhile to query the sizes just as a sanity check, even though this is hard coded for one element type. Might reveal our underlying issue?

Reply:
okay. the gcl unit test doesn't run on device. need to find out why. We probably need to spin up a parallel test suite of this on summit to make sure we don't break cuda too